tools/codegen/dest/lazy_ir.py

```diff
@@ -122,6 +122,11 @@ def __call__(self, f: Union[NativeFunctionsGroup, NativeFunction]) -> List[str]:
     def lowering_function(self, f: Union[NativeFunctionsGroup, NativeFunction]) -> str:
         return ""
 
+    def node_base_ctor_call(self, schema: LazyIrSchema) -> str:
+        # backends can customize the way the node base class constructor is called,
+        # as long as all of its arguments can be generated from information available from the schema
+        return f"{self.node_base}(torch::lazy::OpKind({aten_symbol(schema)})"
```

Comment:
@JackCaoG the way i was thinking you would handle your xla shape computation function is as follows
1. you'd provide a new class that overrides GenLazyIr (GenXLALazyIr) 
2. it would override node_base_ctor_call
  - the schema argument provided to node_base_ctor_call gives you info about all the function arguments
  - you can directly invoke `node_ctor_inputs(schema)` if you want the exact args string that ias passed to node IR constructor; 
  https://github.com/pytorch/pytorch/blob/40f3e85005b60dab26b89c6b09e3fb7cf686d2eb/torchgen/dest/lazy_ir.py#L67
  
   - or you can use an approach like our shape inference/meta kernel
  `', '.join(str(a.name) for a in schema.filtered_args())`
  
  ping me on slack if there is something tricky and hopefully i can help

Reply:
I can overwrite `node_base_ctor_call` but it seems like it is being called in `gen` and the way it worked today is
```
{self.node_base_ctor_call(schema)},
{{{base_ctor_value_args}}}, std::move(shapes),
....
```

If I were to overwrite `node_base_ctor_call` to include the `xla::shape` related info, I would have to make our `XlaNode` constructor looks like
```
Node::Node(torch::lazy::OpKind op, 
                      const std::function<xla::Shape()>& xla_shape_fn
                     OpList operands,
                     const std::function<torch::lazy::Shape()>& shape_fn
```

which is a bit awakward since `xla_shpae_fn` or `xla_shape` would come before the operands.. I guess it is no the end of the world but wondering if there is an easy way for me to change the order of arguments.

Reply:
on a second thought, it might not worth all the new changes, I will change pt/xla constructor for now.

Reply:
oh, i confused myself when I made that change.  My intention was that `node_base_ctor_call` function produces the whole call to the node base ctor.  I must have thought that the other arguments ({{{base_ctor_value_args}}}, std::move(shapes)) are initializer list rather than base class ctor args.  

So, you're right that you can change your ctor and make this work... but I would be happy to fix it instead to do what I originally intended.  basically, i would just put all of the stuff into `node_base_ctor_call` and then you could override it as you like.  Should I proceed with that?

If so, I can bundle that together with other changes you asked for..

Reply:
make `node_base_ctor_call` produce the whole call to the node base ctor would be very nice. If you could bundle this with couple other issues I raised I think I am good to do(for `abs` at least).

Reply:
https://github.com/pytorch/pytorch/pull/76458